### DISCUSSION TITLE
gtl set and maps vs parallel_hashmap

### DISCUSSION THREAD

**Discussion by Nosenzor** [greg7mdp/gtl#27] [General]
Hi Greg,
I've been a happy `parallel_hashmap `user for a year or so. And now i want to use `gtl `for the LRU memoizer. LRU memoizer is a super nice work, thank you ! I've tested on a mac and i could save a lot of computation time using the memoizer.
Then I've tested it on windows and MSVC is complaining about `uint64_t umul128(uint64_t,uint64_t,uint64_t *)` that is defined in both `parallel_hashmap `and `gtl`.
I was thinking to remove parallel_hashmap to use only `gtl ` that also have `hashmaps `, i suspect they are the same map and set but i wanted a confirmation before doing that change.

**Comment by greg7mdp** [comment]
Thanks for the kind words @Nosenzor . I'm glad you like the LRU memoizer. 
Yes, they are the same maps. If you are using `c++20` or above I do recommend using `gtl` instead of `parallel-hashmap`. Mosttly you just change `phmap::` to `gtl::`.
